lib/hydra/derivatives/processors/active_encode.rb

```diff
@@ -5,18 +5,23 @@ class ActiveEncode < Processor
     def process
       encode = ::ActiveEncode::Base.create(source_path, directives)
 
-      # TODO: wait until the encode job succeeds then call output_file_service with the output url
-      # while(encode.reload.running?) do
-      #   p "Wait: #{Time.now}"
-      #   sleep 10
-      # end
+      # Wait until the encoding job is finished
+      # while(encode.reload.running?) { sleep 10 }
 
       raise_exception_if_encoding_failed(encode)
+      raise_exception_if_encoding_cancelled(encode)
+
+      # TODO: call output_file_service with the output url
     end
 
     def raise_exception_if_encoding_failed(encode)
       return unless encode.failed?
-      raise StandardError.new("Encoding failed: #{encode.errors.join(' ; ')}")
+      raise StandardError.new("Encoding failed for #{source_path}: #{encode.errors.join(' ; ')}")
```

Comment:
Yes, I think this would benefit from our own subclass of exception, and that subclass can take the source path as an argument (something like:

```
class EncodingException < RuntimeError
  def initialize(source_path, message)
    ...
  end
end
```
) and use that both to build the message, but also provide the callers with something they could reasonably expect to handle.